File: DayThree/Java/CaeserCypherTest.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class CaeserCypherTest{

    @Test
    public void testThatEncryptTextFunctionReturnsEncryptedText(){

    String word = "Grace";

    int shift = 3;

    String actual = CaeserCypher.encryptText(word,shift);

    String expected = "Judfh";

    assertEquals(actual,expected);
    }

     @Test
    public void testThatDecryptTextFunctionReturnsDecryptedText(){

    String word = "Judfh";

    int shift = 3;

    String actual = CaeserCypher.decryptText(word,shift);

    String expected = "Grace";

    assertEquals(actual,expected);
    }

    @Test
    public void testThatEncryptAndDecrptFunctionReturnsInitalText(){

    String word = "Grace";

    int shift = 3;

    String actual = CaeserCypher.encryptText(word,shift);

    String expected = CaeserCypher.decryptText(actual,shift);

    assertEquals(word,expected);
    }

}

File: DayThree/JavaScript/caeser-cypher.test.js
import {encryptText,decryptText} from "./caeser-cypher.js";

test("test that Encryption Fucnction returns encrypted text",()=>{

    let word = "Grace";

    let shift = 3;

    let encryptedText = "Judfh";

    expect(encryptText(word,shift)) .toBe(encryptedText);
})

test("test that Decryption Fucnction returns decrypted text",()=>{

    let word = "Judfh";

    let shift = 3;

    let decryptedText = "Grace";

    expect(decryptText(word,shift)) .toBe(decryptedText);
})

test("test that Encryption and Decryption Fucnction returns original text",()=>{

    let word = "Grace";

    let shift = 3;

    let encryptedText = encryptText(word,shift);

    expect(decryptText(encryptedText,shift)) .toBe(word);
})
